library/std/src/os/horizon/thread.rs

```diff
@@ -0,0 +1,88 @@
+//! Nintendo 3DS-specific extensions to primitives in the [`std::thread`] module.
+//!
+//! All 3DS models have at least two CPU cores available to spawn threads on:
+//! The application core (appcore) and the system core (syscore). The New 3DS
+//! has an additional two cores, the first of which can also run user-created
+//! threads.
+//!
+//! Threads spawned on the appcore are cooperative rather than preemptive. This
+//! means that threads must explicitly yield control to other threads (whether
+//! via synchronization primitives or explicit calls to `yield_now`) when they
+//! are not actively performing work. Failure to do so may result in control
+//! flow being stuck in an inactive thread while the other threads are powerless
+//! to continue their work.
+//!
+//! However, it is possible to spawn one fully preemptive thread on the syscore
+//! by using a service call to reserve a slice of time for a thread to run.
+//! Attempting to run more than one thread at a time on the syscore will result
+//! in an error.
+//!
+//! [`std::thread`]: crate::thread
+
+#![unstable(feature = "horizon_thread_ext", issue = "none")]
```

Comment:
I looked at creating a tracking issue for this, but held off for now due to this note in the template:
> If the new feature is small, it may be fine to skip the RFC process. In that
case, you can use use `issue = "none"` in your initial implementation PR. The
reviewer will ask you to open a tracking issue if they agree your feature can be
added without an RFC.

So once someone comments that this is an acceptable feature to skip an RFC for, I can create the tracking issue.